assets/js/base/context/cart-checkout/cart/index.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { createContext, useContext } from '@wordpress/element';
+
+/**
+ * @typedef {import('@woocommerce/type-defs/contexts').CartDataContext} CartDataContext
+ */
+
+const CartContext = createContext( {} );
+
+/**
+ * @return {CartDataContext} Returns the checkout data context value
+ */
+export const useCartContext = () => {
+	return useContext( CartContext );
+};
+
+/**
+ * Cart provider
+ *
+ * @param {Object}  props                     Incoming props for the provider.
+ * @param {Array}   props.children            The children being wrapped.
+ * @param {boolean} props.isEditor            Whether the checkout is in the
+ *                                            editor context or not.
+ */
+export const CartProvider = ( { children, isEditor } ) => {
+	/**
+	 * @type {CartDataContext}
+	 */
+	const cartData = {
+		isEditor,
+	};
+	return (
+		<CartContext.Provider value={ cartData }>
+			{ children }
+		</CartContext.Provider>
```

Comment:
Eventually we could implement `ShippingMethodDataProvider` (and potentially `PaymentMethodDataProvider` if/when we support express payment methods in the cart) in here similarly to how it was done for `CheckoutDataProvider`.  But definitely not needed in this pull, nor until we have cause to add it.